gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/UserDefinedFunction.java

```diff
@@ -0,0 +1,151 @@
+package com.google.gcloud.bigquery;
+
+import com.google.api.services.bigquery.model.UserDefinedFunctionResource;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Google BigQuery User Defined Function. BigQuery supports user-defined functions (UDFs) written in
+ * JavaScript. A UDF is similar to the "Map" function in a MapReduce: it takes a single row as input
+ * and produces zero or more rows as output. The output can potentially have a different schema than
+ * the input.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/user-defined-functions">User-Defined Functions</a>
+ */
+public abstract class UserDefinedFunction implements Serializable {
+
+  static final Function<UserDefinedFunctionResource, UserDefinedFunction> FROM_PB_FUNCTION =
+      new Function<UserDefinedFunctionResource, UserDefinedFunction>() {
+        @Override
+        public UserDefinedFunction apply(UserDefinedFunctionResource userDefinedFunctionPb) {
+          return UserDefinedFunction.fromPb(userDefinedFunctionPb);
+        }
+      };
+  static final Function<UserDefinedFunction, UserDefinedFunctionResource> TO_PB_FUNCTION =
+      new Function<UserDefinedFunction, UserDefinedFunctionResource>() {
+        @Override
+        public UserDefinedFunctionResource apply(UserDefinedFunction userDefinedFunction) {
+          return userDefinedFunction.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 8704260561787440287L;
+
+  /**
+   * Type of user-defined function. User defined functions can be provided inline as code blobs
+   * ({@link #INLINE}) or as a Google Cloud Storage URI ({@link #FROM_URI}).
+   */
+  public enum Type {
+    INLINE,
+    FROM_URI
+  }
+
+  private final Type type;
+  private final String functionDefinition;
+
+  UserDefinedFunction(Type type, String functionDefinition) {
+    this.type = type;
+    this.functionDefinition = functionDefinition;
+  }
+
+  public Type type() {
+    return type;
+  }
+
+  /**
+   * Returns function's definition. If {@link #type()} is {@link Type#INLINE} this method returns
+   * a code blob. If {@link #type()} is {@link Type#FROM_URI} this method returns a Google Cloud
+   * Storage URI (e.g. gs://bucket/path).
+   */
+  public String functionDefinition() {
```

Comment:
How about `content` considering the scope `UserDefinedFunction` and the fact that it may actually not be a function definition but rather a reference.